appeals/web/src/server/lib/display-page-formatter.js

```diff
@@ -0,0 +1,116 @@
+import logger from '#lib/logger.js';
+import { buildHtmSpan } from '#lib/nunjucks-template-builders/tag-builders.js';
+export const formatDocumentActionLink = (
+	/** @type {any} */ appealId,
+	/** @type {any} */ listOfDocuments
+) => {
+	// If list of documents is in the correct format
+	if (
+		listOfDocuments != null &&
+		typeof listOfDocuments === 'object' &&
+		'documents' in listOfDocuments
+	) {
+		if (Array.isArray(listOfDocuments.documents) && listOfDocuments.documents.length > 0) {
+			return `/appeals-service/appeal-details/${listOfDocuments.documents[0].caseId}/documents/${listOfDocuments.documents[0].folderId}/upload/${listOfDocuments.documents[0].id}`;
+		}
+		return `/appeals-service/appeal-details/${appealId}/documents/${listOfDocuments.folderId}/upload/`;
+	}
+	return `#`;
+};
+
+export const formatListOfNotificationMethodsToHtml = (
+	/** @type {string | any[]} */ notificationMethods
+) => {
+	let html = ``;
+	if (notificationMethods.length > 0) {
+		for (const method of notificationMethods) {
+			html += `<span>${method.name}</span>`;
+			html += notificationMethods.indexOf(method) !== notificationMethods.length - 1 ? `</br>` : ``;
+		}
+	}
+	return html;
+};
+
+/**
+ * @param {string} answer
+ * @param {string} details
+ */
+export const formatAnswerAndDetails = (answer, details) => {
+	return answer === 'Yes'
+		? `${buildHtmSpan(answer)}<br>${buildHtmSpan(details)}`
+		: `${buildHtmSpan(answer)}`;
+};
+
+export const formatListOfAppeals = (
+	/** @type {{appealId: Number, appealReference: string}[]} */ listOfAppeals
+) => {
+	if (listOfAppeals.length > 0) {
+		let formattedLinks = ``;
+		for (let i = 0; i < listOfAppeals.length; i++) {
+			const appealReferenceFragments = listOfAppeals[i].appealReference.split('/');
+			const shortAppealReference = appealReferenceFragments?.[appealReferenceFragments.length - 1];
+			formattedLinks += `<a href='/appeals-service/appeal-details/${listOfAppeals[i].appealId}'>${shortAppealReference}</a>`;
+			formattedLinks += i + 1 >= listOfAppeals.length ? '' : '<br>';
+		}
+		return `<span>${formattedLinks}</span>`;
+	}
+	return undefined;
+};
+
+/** @returns {string} */
+export const formatListOfListedBuildingNumbers = (
+	/** @type {string | any[]} */ listOfListedBuildingNumbers
+) => {
+	if (listOfListedBuildingNumbers.length > 0) {
+		let formattedLinks = ``;
+		for (let i = 0; i < listOfListedBuildingNumbers.length; i++) {
+			if ('listEntry' in listOfListedBuildingNumbers[i]) {
+				formattedLinks += `<a href='https://historicengland.org.uk/listing/the-list/list-entry/${listOfListedBuildingNumbers[i].listEntry}'>${listOfListedBuildingNumbers[i].listEntry}</a>`;
```

Comment:
Similar to the above, if we had `listOfListedBuildingNumbers[i].listEntry` as a variable, it'd be easier to read.